storage/src/main/java/tech/pegasys/teku/storage/server/Database.java

```diff
@@ -54,6 +55,26 @@ public interface Database extends AutoCloseable {
 
   void updateWeakSubjectivityState(WeakSubjectivityUpdate weakSubjectivityUpdate);
 
+  void storeUnconfirmedBlobsSidecar(BlobsSidecar blobsSidecar);
+
+  void confirmBlobsSidecar(SlotAndBlockRoot slotAndBlockRoot);
+
+  Optional<BlobsSidecar> getBlobsSidecar(SlotAndBlockRoot slotAndBlockRoot);
```

Comment:
So to answer, for example, on `BeaconBlockAndBlobsSidecarByRoot` we will need
1. Query DB for slot for each root
2. Query unconfirmed blobsSidecar for each pair to avoid it
3. Get BlobsSidecar for remaining pairs using this method

not sure that such complex flow is needed.
What are current cases we have with BlobsSidecar:
write:
- gossip receive (coupled, goes to import)
- sync receive (probably coupled too, in discussion, goes to import)

read:
- BeaconBlockAndBlobsSidecarByRoot (root list)
- BlobsSidecarsByRange (slot range)

delete:
- fork pruning (drop BlobsSidecars from noncanonical chain)
- every epoch remove 1 epoch blobsSidecars

Do I miss something or wrong with it? What potential future flows could we have with it?

Reply:
> So to answer, for example, on BeaconBlockAndBlobsSidecarByRoot we will need

since we have to respond with a coupled block and blobs, we will start looking up blocks first. Once we have an existing blockRoot we can lookup in blobsSidecar without checking the unconfirmed column.

>delete:
>- fork pruning (drop BlobsSidecars from noncanonical chain)
>- every epoch remove 1 epoch blobsSidecars

there is one more. The sequence will be:
- prune all blobs that has not being confirmed (related to blocks that remain in pending\future until discarded - ie never actually imported (successfully or with error))
- fork pruning (pruning blobs for noncanonical blocks up to the most recent finalized slot)
- prune blobs outside the data availability window (regardless finalization or not, just to avoid db overload when chain is not finalizing)